ResearchKit/Common/ORKDiscreteGraphView.m

```diff
@@ -0,0 +1,1023 @@
+        thumbSize = CGSizeMake(15, 15);
+    } else{
+        thumbSize = CGSizeMake(10, 10);
+    }
+    
+    return thumbSize;
+}
+
+
+#pragma mark - View Layout
+
+- (void)layoutSubviews {
+    [super layoutSubviews];
+    
+    CGFloat yAxisPadding = CGRectGetWidth(self.frame)*kYAxisPaddingFactor;
+    
+    //Basic Views
+    
+    self.plotsView.frame = CGRectMake(kORKGraphLeftPadding, kORKGraphTopPadding, CGRectGetWidth(self.frame) - yAxisPadding - kORKGraphLeftPadding, CGRectGetHeight(self.frame) - kXAxisHeight - kORKGraphTopPadding);
+    
+    if (self.emptyLabel) {
+        self.emptyLabel.frame = CGRectMake(kORKGraphLeftPadding, kORKGraphTopPadding, CGRectGetWidth(self.frame) - kORKGraphLeftPadding, CGRectGetHeight(self.frame) - kXAxisHeight - kORKGraphTopPadding);
+    }
+    
+    //Scrubber Views
+    self.scrubberLine.frame = CGRectMake(CGRectGetMinX(self.scrubberLine.frame), kORKGraphTopPadding, 1, CGRectGetHeight(self.plotsView.frame));
+    [self updateScrubberLabel];
+    self.scrubberThumbView.frame = CGRectMake(CGRectGetMinX(self.scrubberThumbView.frame), CGRectGetMinY(self.scrubberThumbView.frame), [self scrubberThumbSize].width, [self scrubberThumbSize].height);
+    self.scrubberThumbView.layer.cornerRadius = self.scrubberThumbView.bounds.size.height/2;
+    
+    [self.xAxisView layoutSubviews];
+}
+
+- (void)refreshGraph {
+    //Clear subviews and sublayers
+    [self.plotsView.layer.sublayers makeObjectsPerformSelector:@selector(removeAllAnimations)];
+    [self.plotsView.layer.sublayers makeObjectsPerformSelector:@selector(removeFromSuperlayer)];
+    
+    [self drawXAxis];
+    [self drawYAxis];
+    
+    [self drawhorizontalReferenceLines];
+    
+    if (self.showsVerticalReferenceLines) {
+        [self drawVerticalReferenceLines];
+    }
+    
+    [self calculateXAxisPoints];
+    
+    [self.dots removeAllObjects];
+    [self.pathLines removeAllObjects];
+    
+    for (int i=0; i<[self numberOfPlots]; i++) {
+        if ([self numberOfPointsinPlot:i] <= 1) {
+            return;
+        } else {
+            [self drawGraphForPlotIndex:i];
+        }
+    }
+    
+    if (!self.hasDataPoint) {
+        [self setupEmptyView];
+    } else {
+        if (self.emptyLabel) {
+            [self.emptyLabel removeFromSuperview];
+        }
+    }
+    
+    [self animateLayersSequentially];
+}
+
+- (void)setupEmptyView {
+    if (!_emptyLabel) {
+        
+        _emptyLabel = [[UILabel alloc] initWithFrame:CGRectMake(kORKGraphLeftPadding, kORKGraphTopPadding, CGRectGetWidth(self.frame) - kORKGraphLeftPadding, CGRectGetHeight(self.frame) - kXAxisHeight - kORKGraphTopPadding)];
+        _emptyLabel.text = self.emptyText;
+        _emptyLabel.textAlignment = NSTextAlignmentCenter;
+        _emptyLabel.font = [UIFont fontWithName:@"Helvetica" size:25];
+        _emptyLabel.textColor = [UIColor lightGrayColor];
+    }
+    
+    [self addSubview:_emptyLabel];
+}
+
+#pragma mark - Data
```

Comment:
If we make the common *data source protocol*, we'll be able to reuse code here as well.